pkg/operator/ceph/cluster/cluster.go

```diff
@@ -595,28 +600,86 @@ func (c *cluster) reportTelemetry() {
 	telemetry.ReportKeyValue(c.context, c.ClusterInfo, telemetry.ExternalModeEnabledKey, strconv.FormatBool(c.Spec.External.Enable))
 }
 
+func (c *cluster) generateMonAssimilateConf(clientName string, settings map[string]string) error {
+	root := path.Join(c.context.ConfigDir, c.Namespace)
+	assimilateConfPath := path.Join(root, fmt.Sprintf("%s.txt", clientName))
+	if err := os.MkdirAll(root, 0744); err != nil {
+		return errors.Wrapf(err, "failed to create assimilateConfPath for %s.", clientName)
+	}
+
+	err := ioutil.WriteFile(assimilateConfPath, []byte(""), 0600)
+	if err != nil {
+		rook.TerminateFatal(errors.Wrapf(err, "failed to write config file"))
+	}
+
+	defer func() {
+		err := os.Remove(assimilateConfPath)
+		if err != nil {
+			logger.Errorf("failed to remove file %q. %v", assimilateConfPath, err)
+		}
+	}()
+
+	configFile := ini.Empty()
+	s, err := configFile.NewSection(clientName)
+	if err != nil {
+		return err
+	}
+
+	for key, val := range settings {
+		if _, err := s.NewKey(key, val); err != nil {
+			return errors.Wrapf(err, "failed to add key %s", key)
+		}
+	}
+	if err := configFile.SaveTo(assimilateConfPath); err != nil {
+		return errors.Wrapf(err, "failed to save config file %s", assimilateConfPath)
+	}
+
+	fileContent, err := ioutil.ReadFile(assimilateConfPath)
+	if err != nil {
+		logger.Errorf("failed to open assimilate input file %s. %c", assimilateConfPath, err)
+	}
+	logger.Infof("applying assimilate-conf setting %s", string(fileContent))
+
+	args := []string{"config", "assimilate-conf", "-i", assimilateConfPath, "-o", assimilateConfPath + ".out"}
+	cephCmd := client.NewCephCommand(c.context, c.ClusterInfo, args)
+	out, err := cephCmd.RunWithTimeout(pkgexec.CephCommandsTimeout)
+	if err != nil {
+		logger.Errorf("failed to run command ceph %s", args)
+
+		fileContent, err := ioutil.ReadFile(assimilateConfPath + ".out")
+		if err != nil {
+			logger.Errorf("failed to open assimilate output file %s.out %v", assimilateConfPath, err)
+		}
+		logger.Infof("failed assimilate-conf setting %s", string(fileContent))
+
+		return errors.Wrapf(err, "failed to set ceph config in the centralized mon configuration database; "+
+			"you may need to use the rook-config-override ConfigMap. output: %s", string(out))
+	}
+	logger.Infof("successfully set %s option to the mon configuration database", settings)
```

Comment:
No need for the settings here since they were dumped to the log above
```suggestion
	logger.Info("successfully applied settings to the mon configuration database")
```

Reply:
done